design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -189,19 +175,34 @@ The invoice entity will have the similar to the order schema:
 type Invoice {
     id: ID! @doc("the ID of the invoice, used for API purposes")
     number: String! @doc("sequential invoice number")
-    total: InvoiceTotal! @doc("invoice total amount details")
-    items: [InvoiceItem]! @doc("invoiced product details")
+    total: InvoiceTotal @doc("invoice total amount details")
+    items: [InvoiceItemInterface] @doc("invoiced product details")
+    comments: [CommentItem] @doc("comments on the invoice")
 }
 
 @doc("Invoice item details")
-type InvoiceItem implements SalesItemInterface{
-    quantity_invoiced: Float! @doc("number of invoiced items")
+type InvoiceItemInterface {
+    id: ID! @doc("invoice item unique identifier") #base64encode(invoiceItemId)
+    order_item_id: String @doc("link to order item")
```

Comment:
Is there a reason to prefer `order_item_id` here and not something like `OrderItemInterface`.

Typically in a GraphQL schema, anywhere you _would_ put a foreign key to an entity, it would make more sense to allow selecting that entity. Otherwise, it can force the client-side into needing multiple round trips for the data

Reply:
It seems preferable to return an id for use as a reference rather than returning the object here. We don't know exactly how this will be used, but it seems like it would not be preferable to query the order items on the order level and then also on each level of InvoiceItems, ShipmentItems, etc. However it does seem likely that having this reference would be useful, though I don't have concrete examples.

Reply:
If someone didn't want the whole object, wouldn't they just not query for it though?

Reply:
We can return the whole item.
@joni-jones is that alright with you? I know you had concern with that complicating the schema

Reply:
I don't know the real use case where order item data is needed in the scope of invoice/shipment/credit memo item. The `order_item_id` would help to referer to the original order item if it needed (some custom logic to show which order items shipped/invoiced/refunded), I don't think the whole data would be useful.

Reply:
What about if a UI dev wants to make an `a` tag pointing to the URL of the product? Wouldn't a second round trip be needed for `product_url_key`?

Reply:
@DrewML @joni-jones I do see a need to refer back to the Order Item data for getting the selected and entered options. based on the above screenshot as well(" 1 X Sprite..") which the InvoiceItemInterface doesn't offer.

Reply:
Discussed pros and cons of this offline. We decided to return order_item instead of order_item_id, it may introduce complexity and performance implications on frontend. But we prioritize our API's alignment with standard GraphQL approach.